File: Ressurser/New_Steiner_Algo/steiner_print.m
function [] = steiner_print(G)
% function [] = steiner_print(G)
% print the steiner minimum-spanning-tree after computing 
%   the steiner tree with the function Steiner_Tree
%
% Reggie.Davidrajuh@uis.no (c) August 2011

TotalWT = 0;
A2 = G.Steiner;

disp(' ');
disp('Printing the edges of Steiner tree ... ');

for i = 1 : length(G.nodes)-1
    for j = i+1 : length(G.nodes)
        if A2(i, j)
            wt = A2(i, j);
            disp(['Edge-', int2str(i), ':  ', G.nodes(i).name, '-',...
                  G.nodes(j).name, '    Wt: ', int2str(wt)]);
            TotalWT = TotalWT + wt;
        end
    end    
end

% finally, display the total weight
disp(['Total Weight : ', int2str(TotalWT)]);   

File: Kode/steiner_print.py
import numpy as np;
from sample_graph import SampleGraph;


def steiner_print(Steiner):
    TotalWT = 0
    A2 = Steiner.A
    print(' ')
    print("Printing the edges of Steiner tree ...")
    print(f"Terminal nodes: {Steiner.Terminals}")
    for i in range(len(Steiner.Nodes)-1):
        for j in range(i+1,len(Steiner.Nodes)):
            if A2[i][j]: 
                wt = A2[i][j]
                print(f"Edge-{i}:{Steiner.Nodes[i]}-{Steiner.Nodes[j]}    Wt: {wt}")
                TotalWT += wt

           

    print(f"Total Weight: {TotalWT}")